app/modules/code_provider/tests/code_provider_service_test.py

```
@@ -0,0 +1,177 @@
+    service = Mock(spec=LocalRepoService)
+    service.get_repo.return_value = MagicMock()
+    service.get_project_structure_async.return_value = {"files": [], "directories": []}
+    service.get_file_content.return_value = "test content"
+    service.project_manager = mock_project_manager
+    return service
+
+class TestCodeProviderService:
+    
+    def test_init_development_mode(self, mock_sql_db):
+        with patch.dict(os.environ, {"isDevelopmentMode": "enabled"}):
+            with patch("app.modules.code_provider.code_provider_service.LocalRepoService") as mock_local:
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)